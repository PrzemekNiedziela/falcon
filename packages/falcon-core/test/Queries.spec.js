// @flow
/* eslint no-await-in-loop: 0 */
import Queries from '../src/config/QueryManager';

async function queryFactory(queries, queryCount: number = 1) {
  const array = new Array(queryCount);
  for (let i = 0; i < array.length; i++) {
    array[i] = await queries.add({
      id: `test-id-${i + 1}`,
      name: `test-query-${i + 1}`,
      type: 'sqlite',
      query: 'SELECT * FROM users',
      color: 'default'
    });
  }
  return Promise.all(array);
}

describe('Queries', function testQueries() {
  beforeEach(async () => {
    this.queries = new Queries();
    await queryFactory(this.queries, 3);
  });

  afterEach(async () => {
    await this.queries.removeAll();
  });

  it('should get all queries', async () => {
    const queries = await this.queries.getAll();
    expect(queries).toMatchSnapshot();
  });

  it('should delete a single query', async () => {
    const queries = await this.queries.getAll();
    const queryIdToDelete = queries[0].id;
    await this.queries.remove(queryIdToDelete);
    const newQueries = await this.queries.getAll();
    expect(newQueries).toMatchSnapshot();
  });

  it('should update a single query', async () => {
    const queries = await this.queries.getAll();
    const queryIdToDelete = queries[0].id;
    await this.queries.update(queryIdToDelete, {
      id: 'test-id-foo',
      name: 'test-query-foo',
      password: 'test-password',
      type: 'sqlite'
    });
    const newQueries = await this.queries.getAll();
    expect(newQueries).toMatchSnapshot();
  });

  it('should get a single query', async () => {
    const queries = await this.queries.getAll();
    const queryId = queries[2].id;
    const query = await this.queries.get(queryId);
    expect(query).toMatchSnapshot();
  });

  it('should perform basic validation', async () => {
    expect(() => {
      this.queries.validateBeforeCreation({
        id: 12,
        type: 'sqlite'
      });
    }).toThrowErrorMatchingSnapshot();
    expect(() => {
      this.queries.validateBeforeCreation({
        id: 12,
        type: 'sqlite'
      });
    }).toThrowErrorMatchingSnapshot();
    expect(() => {
      this.queries.validateBeforeCreation({
        id: 'foo',
        type: 'sqlite'
      });
    }).toThrowErrorMatchingSnapshot();
    expect(() => {
      this.queries.validateBeforeCreation({
        id: 'foo',
        name: 'foo',
        type: 'sqlite'
      });
    }).toThrowErrorMatchingSnapshot();
    expect(() => {
      this.queries.validateBeforeCreation({
        id: 'foo',
        name: 'foo',
        type: 'sqlite'
      });
    }).toThrowErrorMatchingSnapshot();
  });
});
