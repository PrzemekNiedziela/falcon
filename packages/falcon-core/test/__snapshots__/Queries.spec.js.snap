// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Queries should delete a single query 1`] = `
Array [
  Object {
    "color": "default",
    "id": "test-id-2",
    "name": "test-query-2",
    "query": "SELECT * FROM users",
    "type": "sqlite",
  },
  Object {
    "color": "default",
    "id": "test-id-3",
    "name": "test-query-3",
    "query": "SELECT * FROM users",
    "type": "sqlite",
  },
]
`;

exports[`Queries should get a single query 1`] = `
Object {
  "color": "default",
  "id": "test-id-3",
  "name": "test-query-3",
  "query": "SELECT * FROM users",
  "type": "sqlite",
}
`;

exports[`Queries should get all queries 1`] = `
Array [
  Object {
    "color": "default",
    "id": "test-id-1",
    "name": "test-query-1",
    "query": "SELECT * FROM users",
    "type": "sqlite",
  },
  Object {
    "color": "default",
    "id": "test-id-2",
    "name": "test-query-2",
    "query": "SELECT * FROM users",
    "type": "sqlite",
  },
  Object {
    "color": "default",
    "id": "test-id-3",
    "name": "test-query-3",
    "query": "SELECT * FROM users",
    "type": "sqlite",
  },
]
`;

exports[`Queries should perform basic validation 1`] = `"Failed validation: [{\\"message\\":\\"\\\\\\"id\\\\\\" must be a string\\",\\"fieldName\\":\\"id\\"},{\\"message\\":\\"\\\\\\"name\\\\\\" is required\\",\\"fieldName\\":\\"name\\"},{\\"message\\":\\"\\\\\\"query\\\\\\" is required\\",\\"fieldName\\":\\"query\\"}]"`;

exports[`Queries should perform basic validation 2`] = `"Failed validation: [{\\"message\\":\\"\\\\\\"id\\\\\\" must be a string\\",\\"fieldName\\":\\"id\\"},{\\"message\\":\\"\\\\\\"name\\\\\\" is required\\",\\"fieldName\\":\\"name\\"},{\\"message\\":\\"\\\\\\"query\\\\\\" is required\\",\\"fieldName\\":\\"query\\"}]"`;

exports[`Queries should perform basic validation 3`] = `"Failed validation: [{\\"message\\":\\"\\\\\\"name\\\\\\" is required\\",\\"fieldName\\":\\"name\\"},{\\"message\\":\\"\\\\\\"query\\\\\\" is required\\",\\"fieldName\\":\\"query\\"}]"`;

exports[`Queries should perform basic validation 4`] = `"Failed validation: [{\\"message\\":\\"\\\\\\"query\\\\\\" is required\\",\\"fieldName\\":\\"query\\"}]"`;

exports[`Queries should perform basic validation 5`] = `"Failed validation: [{\\"message\\":\\"\\\\\\"query\\\\\\" is required\\",\\"fieldName\\":\\"query\\"}]"`;

exports[`Queries should update a single query 1`] = `
Array [
  Object {
    "color": "default",
    "id": "test-id-1",
    "name": "test-query-1",
    "query": "SELECT * FROM users",
    "type": "sqlite",
  },
  Object {
    "color": "default",
    "id": "test-id-2",
    "name": "test-query-2",
    "query": "SELECT * FROM users",
    "type": "sqlite",
  },
  Object {
    "color": "default",
    "id": "test-id-3",
    "name": "test-query-3",
    "query": "SELECT * FROM users",
    "type": "sqlite",
  },
]
`;
